r2/r2/lib/providers/search/solrsearch.py

```diff
@@ -0,0 +1,742 @@
+# The contents of this file are subject to the Common Public Attribution
+# License Version 1.0. (the "License"); you may not use this file except in
+# compliance with the License. You may obtain a copy of the License at
+# http://code.reddit.com/LICENSE. The License is based on the Mozilla Public
+# License Version 1.1, but Sections 14 and 15 have been added to cover use of
+# software over a computer network and provide for limited attribution for the
+# Original Developer. In addition, Exhibit A has been modified to be consistent
+# with Exhibit B.
+#
+# Software distributed under the License is distributed on an "AS IS" basis,
+# WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+# the specific language governing rights and limitations under the License.
+#
+# The Original Code is reddit.
+#
+# The Original Developer is the Initial Developer.  The Initial Developer of
+# the Original Code is reddit Inc.
+#
+# All portions of the code written by reddit are Copyright (c) 2006-2015 reddit
+# Inc. All Rights Reserved.
+###############################################################################
+
+from datetime import datetime, timedelta
+import cPickle as pickle
+import functools
+import httplib
+import urllib
+import json
+import time
+import re
+
+
+from lxml import etree
+from r2.lib import amqp, filters
+from r2.lib.db.operators import desc
+from r2.lib.db.sorts import epoch_seconds
+from r2.lib.configparse import ConfigValue
+import r2.lib.utils as r2utils
+from r2.models import (Account, Link, Subreddit, All, DefaultSR,
+                       MultiReddit, DomainSR, Friends, ModContribSR,
+                       FakeSubreddit, NotFound)
+from pylons import g, c
+
+from r2.lib.providers.search.base import safe_get, InvalidQuery, SearchHTTPError, \
+        LinkFields, SubredditFields, Results
+from r2.lib.providers.search import SearchProvider
```

Comment:
Please order the imports according to PEP 8 and our styleguide: https://github.com/reddit/styleguide/tree/master/python

Specifically, standard lib imports first, then a blank line, then 3rd party imports (such as pylons), then a blank line, then r2 imports. Each group should be alphabetized.